### PR TITLE
Migrate partners to Payload

### LINKED ISSUE #563: @/codeforafrica partners
Implement Partners

The implementation should also keep in mind https://github.com/CodeForAfrica/ui/pull/443 since this approach promises to minimise changes should content editors move blocks around.

Current content:
https://github.com/CodeForAfrica/ui/tree/main/apps/codeforafrica/content/partners

### PR DESCRIPTION
**Pull request by koechkevin**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Single Partner page to be implemented in #582 

Fixes #563

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Screenshots
image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
Your reviews @kelvinkipruto / @saminegash

**Review by kelvinkipruto — APPROVED** [review]
LGTM

File: apps/codeforafrica/src/payload/collections/Partners.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      relationTo: "media",
+      label: {
+        en: "Logo",
+        pt: "Imagem",
+        fr: "Image",
+      },
+      filterOptions: {
+        mimeType: { contains: "image" },
+      },
+    },
```
Comment (kilemensi) [inline apps/codeforafrica/src/payload/collections/Partners.js:54]: I think we have `image` field now.

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**